### PR TITLE
[KOGITO-2282] - Shared libaries: PR creation

### PR DESCRIPTION
**Pull request by Kaitou786**
See: https://issues.redhat.com/browse/KOGITO-2282
Signed-off-by: Tarun Khandelwal <tarkhand@redhat.com>

### CODE REVIEW

File: vars/githubscm.groovy [hunk 1]
```diff
@@ -74,6 +74,27 @@ def mergeSourceIntoTarget(String repository, String sourceAuthor, String sourceB
     """
 }
 
+def forkRepo(String credentialID){
+    withCredentials([usernamePassword(credentialsId: credentialID, usernameVariable: 'GITHUB_USER', passwordVariable: 'GITHUB_PASSWORD')]){
+    sh 'git config --global hub.protocol https'
```
Comment (Ginxo) [inline vars/githubscm.groovy:79]: Indentation please

File: vars/githubscm.groovy [hunk 2]
```diff
@@ -74,6 +74,27 @@ def mergeSourceIntoTarget(String repository, String sourceAuthor, String sourceB
...
+    sh "hub fork --remote-name=origin"
+    sh 'git remote -v'
+    }
+}
+
+def createPR(String credentialID, String targetBranch, String PRMessage){
```
Comment (Ginxo) [inline vars/githubscm.groovy:85]: two things space after `)`
I would do
`def createPR(String targetBranch, String PRMessage, String credentialID = 'kie-ci'){`

File: vars/githubscm.groovy [hunk 3]
```diff
@@ -74,6 +74,27 @@ def mergeSourceIntoTarget(String repository, String sourceAuthor, String sourceB
...
+    withCredentials([usernamePassword(credentialsId: credentialID, usernameVariable: 'GITHUB_USER', passwordVariable: 'GITHUB_PASSWORD')]){
+        try{
+            def PRLink = sh(returnStdout: true, script: "hub pull-request -m '${PRMessage}' -b '${targetBranch}' ")
```
Comment (Ginxo) [inline vars/githubscm.groovy:88]: `pullRequestLink` instead No uppercase for the first character

File: vars/githubscm.groovy [hunk 4]
```diff
@@ -74,6 +74,27 @@ def mergeSourceIntoTarget(String repository, String sourceAuthor, String sourceB
...
+        } catch catch (Exception e) {
+            println "[ERROR] Unable to create PR make sure the targetBranch ${targetBranch} is correct"
+            throw e;
+        }
+    println "Please see the created PR at: ${PRLink}"
```
Comment (Ginxo) [inline vars/githubscm.groovy:93]: indentation

File: vars/githubscm.groovy [hunk 5]
```diff
@@ -74,6 +74,27 @@ def mergeSourceIntoTarget(String repository, String sourceAuthor, String sourceB
...
+        sh 'git config --global hub.protocol https'
+        sh "hub fork --remote-name=origin"
+        sh 'git remote -v'
+    }
+}
+
+def createPR(String targetBranch, String pullRequestMessage, String credentialID='kie-ci') {
+    withCredentials([usernamePassword(credentialsId: credentialID, usernameVariable: 'GITHUB_USER', passwordVariable: 'GITHUB_PASSWORD')]){
+        try{
+            def pullRequestLink = sh(returnStdout: true, script: "hub pull-request -m '${pullRequestMessage}' -b '${targetBranch}' ")
```
Comment (radtriste) [inline vars/githubscm.groovy:88]: please add a `.trim()` to the output of the shell command

Comment (Kaitou786) [reply vars/githubscm.groovy]: Done

**Comment by Kaitou786** [comment]
updated `targetBranch` to defaults to `master`

**Pull request merged**